packages/suite/src/views/wallet/send/Outputs/TokenSelect/TokenSelect.tsx

```diff
@@ -80,18 +80,18 @@ export const TokenSelect = ({ outputId }: TokenSelectProps) => {
     }, [account, outputId, tokenWatch, setAmount, getValues, isSetMaxActive, isDebugModeActive]);
 
     useEffect(() => {
-        if (sendFormPrefill) {
-            setValue(tokenInputName, sendFormPrefill, { shouldValidate: true, shouldDirty: true });
+        if (sendFormPrefillContractAddress) {
+            setValue(tokenInputName, sendFormPrefillContractAddress, {
+                shouldValidate: true,
+                shouldDirty: true,
+            });
             setDraftSaveRequest(true);
 
             return () => {
-                dispatch({
-                    type: SUITE.SET_SEND_FORM_PREFILL,
-                    payload: '',
-                });
+                dispatch(setSendFormPrefill({ contractAddress: undefined }));
             };
         }
-    }, [sendFormPrefill, setValue, tokenInputName, setDraftSaveRequest, dispatch]);
+    }, [sendFormPrefillContractAddress, setValue, tokenInputName, setDraftSaveRequest, dispatch]);
```

Comment:
Nothing interesting here, Just use `setSendFormPrefill` action instead of an inline object